cmd/serve.go

```diff
@@ -64,27 +66,34 @@ func runServe(_ *cobra.Command, _ []string) {
 	}
 
 	if serveMetrics != "" {
-		reg = prometheus.NewRegistry()
-
-		siloMetrics = siloprom.New(reg, siloprom.DefaultConfig())
-
-		// Add the default go metrics
-		reg.MustRegister(
-			collectors.NewGoCollector(),
-			collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
-		)
-
-		http.Handle("/metrics", promhttp.HandlerFor(
-			reg,
-			promhttp.HandlerOpts{
-				// Opt into OpenMetrics to support exemplars.
-				EnableOpenMetrics: true,
-				// Pass custom registry
-				Registry: reg,
-			},
-		))
-
-		go http.ListenAndServe(serveMetrics, nil)
+		switch serveMetricsSystem {
+		case "prometheus":
```

Comment:
Might be worthwhile to use an enum for this instead of magic strings, e.g
```go
type MetricsSystem string

const (
	MetricsPrometheus MetricsSystem = "prometheus"
	MetricsStatsD     MetricsSystem = "statsd"
)
```
That way the set of valid values is centralized, we avoid typos like "promethues", IDEs can autocomplete valid options, and it becomes straightforward to validate or extend in the future by just adding another constant and case.